src/cmd/services/m3coordinator/server/m3msg/handler.go

```diff
@@ -32,6 +34,8 @@ import (
 	"github.com/uber-go/tally"
 )
 
+var ctx = context.TODO()
```

Comment:
Instead of making this global, can you give one to each handler?  

Also it makes more sense to derive from `context.Background()` rather than `context.TODO()` since we want to signify this was from a background listen server, rather than intending to replace it later.